### DISCUSSION TITLE
How to plot node features. Could not find in docs.

### DISCUSSION THREAD

**Discussion by rozeappletree** [networkx/networkx#5318] [Q&A] [answered]
Hi currenly I am using this code to plot graph
```
  nx.draw(nx_graph, pos=pos, with_labels=True, node_size=500, font_color='yellow')
  nx.draw_networkx_edge_labels(nx_graph, pos=pos, font_color='red')
```
![image](https://user-images.githubusercontent.com/45640029/153486930-b6b0fbbe-3493-4df2-8c68-4cf424c29dcd.png)

But, I need to plot features related to node0, node1, node2 and node3 on the graph also. How may I do it?

**Comment by rossbar** [comment, marked as answer]
Generally, the best way to find a specific visualization feature is to look at the [networkx example gallery](https://networkx.org/documentation/latest/auto_examples/index.html). For example, the first example I noticed with labeled nodes was the [chess example](https://networkx.org/documentation/latest/auto_examples/index.html) which demonstrates how to use `draw_networkx_labels` to draw custom node labels.

**Comment by rozeappletree** [comment]
I tried to use the same but could get any output for it.
I am using  ```nx_graph.add_node(node_index, node_feature)``` to create graph.

The node index and node features are as here:
```
adding node_index: 0 node_features: [0.]
adding node_index: 1 node_features: [2.]
adding node_index: 2 node_features: [4.]
adding node_index: 3 node_features: [6.]
```
But the graphs are same (the one with bbox is with `draw_networkx_labels`)
```
  nx.draw(nx_graph, pos=pos, with_labels=True, node_size=500, font_color='yellow')
  nx.draw_networkx_edge_labels(nx_graph, pos=pos, font_color='red')
  plt.show()
  nx.draw(nx_graph, pos=pos, with_labels=True, node_size=500, font_color='yellow')
  nx.draw_networkx_labels(nx_graph, pos=pos, font_size=14, font_color='red', bbox={'ec':'k', 'fc':'white', 'alpha': 0.7})
  plt.show()
```
![image](https://user-images.githubusercontent.com/45640029/153492345-79ab082b-c4c5-4f39-b805-14376b463c02.png)
![image](https://user-images.githubusercontent.com/45640029/153492362-a291d8eb-0921-4424-b89c-10f672663413.png)

> note that in circles I want node index and in squares I want node features
---
reproduced the plots in this notebook: https://colab.research.google.com/drive/1FWeRjlcuSK1lUhEcV36F2L-ZWXHD0Rvg?usp=sharing

**Reply by rossbar** [reply]
In the call to `nx.draw_networkx_labels` you're not actually passing in a `labels` dict, so the default (i.e. the node id) is being used. Take a closer look at the `nx.draw_networkx_labels` docstring to ensure you're passing in the info you want displayed.

**Reply by rozeappletree** [reply]
Worked thx :)
